src/debugger/rnDebugSession.ts

```diff
@@ -227,6 +275,31 @@ export class RNDebugSession extends LoggingDebugSession {
             return Q.resolve<void>(void 0);
         }
     }
+
+    private handleStartDebugSession(debugSession: vscode.DebugSession) {
+        if (debugSession.type === "pwa-node") {
+            this.nodeSession = debugSession;
+        }
+    }
+
+    private handleTerminateDebugSession(debugSession: vscode.DebugSession) {
+        if (this.reloading && this.reloadingResolve) {
+            this.reloadingResolve();
+        }
+    }
+
+    private clearReloading(): void {
+        if (this.reloading) {
```

Comment:
I suppose that we can create a enum that will represent the status of the debug session and a SessionStatus variable that will show us in what state the debugger is.